packages/testing-utils/src/index.js

```diff
@@ -19,19 +20,31 @@ export const cloneableGenerator = generatorFunc => (...args) => {
 }
 
 export function createMockTask() {
-  let _isRunning = true
-  let _result
-  let _error
+  let status = RUNNING
+  let taskResult
+  let taskError
 
   return {
     [TASK]: true,
-    isRunning: () => _isRunning,
-    result: () => _result,
-    error: () => _error,
+    isRunning: () => status === RUNNING,
+    isCancelled: () => status === CANCELLED,
+    isAborted: () => status === ABORTED,
+    result: () => taskResult,
+    error: () => taskError,
     cancel: () => {},
+    joiners: [],
 
-    setRunning: b => (_isRunning = b),
-    setResult: r => (_result = r),
-    setError: e => (_error = e),
+    setRunning: () => (status = RUNNING),
```

Comment:
just a random thought without full analysis - perhaps `setRunning` should just be deprecated? it seems to me that making a mock task forcefully take a transition that isn't possible in the real code is asking for trouble.

Valid transitions are those:
- `running` -> `done` (can be achieved by calling `setResult`)
- `running` -> `aborted` (can be done by calling `setError`)
- `running` -> `canceled` (could be done by `cancel`)

Thoughts?

Reply:
I agree, I think that's a good idea.

Went ahead and added checks for that and tests to cover it :)